ship config.yml to /usr/share and copy to user home ~/.calyptos

mbacchi opened the pull request:
This changes how we use the config.yml. Instead of placing in /etc
we will put it in /usr/share/calyptos/etc on install, then copy to
~/.calyptos in the user home dir when calyptos is executed.

@viglesiasce please also let me know if you prefer another way to call locate_cfgfile(), I'm not sure how these bootstrap type things are done in Python typically.  Without a class to run an **init**()  method I just stuck that method in main.

mbacchi committed:
ship config.yml to /usr/share and copy to user home ~/.calyptos

This changes how we use the config.yml. Instead of placing in /etc
we will put it in /usr/share/calyptos/etc on install, then copy to
~/.calyptos in the user home dir when calyptos is executed.

viglesiasce commented on bin/calyptos:
```diff
@@ -3,8 +3,28 @@ import argparse
+import shutil
+
+dot_dir = os.path.expanduser('~/.calyptos')
+default_cfgfile = os.path.join(dot_dir + "/" + "config.yml")
```
when using os.path.join its intended to be used as follows:

`default_cfgfile = os.path.join(dot_dir, "config.yml")`

Its functionality is to join add those /'s for you intelligently (ie removing duplicate /'s and such)

viglesiasce commented on bin/calyptos:
```diff
@@ -3,8 +3,28 @@ import argparse
+default_cfgfile = os.path.join(dot_dir + "/" + "config.yml")
+
+def locate_cfgfile():
+    if not os.path.isfile(default_cfgfile):
```
I would create an array of paths to check, then iterate over them in order until I found one and use that as the default.

viglesiasce commented on setup.py:
```diff
@@ -140,6 +140,6 @@ def make_release_tree(self, base_dir, files):
     },
-    #data_files=[('/etc/calyptos/', ['etc/config.yml']),
-    #            ('/usr/share/calyptos/examples/', example_items)],
+    data_files=[('/usr/share/calyptos/etc', ['etc/config.yml']),
```
id like this to be /usr/share/calyptos/ rather than /usr/share/calyptos/etc

Therefore, on CentOS 6 the file will live in `/usr/share/calyptos/config.yml` and on mac/dev systems `~/.calyptos/config.yml`

mbacchi committed:
iterate through possible config file locations

incorporate other PR comments from Vic:
1. use os.path.join properly
2. put config.yml in /usr/share/calyptos

this also copies config.yml to the ~/.calyptos directory
if there IS NOT a calyptos RPM package installed. this
indicates that the user has used setup.py to install as
opposed to rpm/yum, and we are allowing for the possibility
that in a dev environment they may want to edit the config.

mbacchi commented:
@viglesiasce I added your suggestions, they are in commit:

https://github.com/mbacchi/calyptos/commit/c878e0ef3fca17957b35af81cc81c4b9ebcbf9a7

Let me know what you think

viglesiasce commented:
Thanks @mbacchi !!!
